compose/Dockerfile.source

```
@@ -0,0 +1,266 @@
+
+# Build testnet configuration files
+RUN uv run python3 genesis-cli.py testnet.yaml -o /tmp/testnet -c generate
+
+# Remove dynamic topology.json
+RUN find /tmp/testnet -type f -name 'topology.json' -exec rm -f '{}' ';'
+
+#---------------------------------------------------------------------
+
+FROM docker.io/debian:stable-slim AS main
+
+# Set environment variables
+ENV CARDANO_NODE_SOCKET_PATH="/opt/cardano-node/data/db/node.socket"
+
+# Set time zone
+ENV TZ="UTC"
+RUN ln -snf /usr/share/zoneinfo/${TZ} /etc/localtime && \
+    echo ${TZ} > /etc/timezone
+
+# Install packages required in main stage
+RUN apt update && \
+    apt install -y --no-install-recommends \
+        dnsutils \
+        iproute2 \
+        iputils-ping \
+        jq \
+        less \
+        lsof \
+        netbase \
+        netcat-openbsd \
+        openssl \
+        procps \
+        tcpdump \
+        telnet \
+        vim
+
+# Copy libsodium library
+COPY --from=build --chown=root:root /usr/local/lib/libsodium.so /usr/local/lib
+RUN ln -snf /usr/local/lib/libsodium.so /usr/local/lib/libsodium.so.23 && \
+    ln -snf /usr/local/lib/libsodium.so /usr/local/lib/libsodium.so.23.3.0
+RUN ldconfig
+
+# Copy secp256k1 library
+COPY --from=build --chown=root:root /usr/local/src/secp256k1/.libs/libsecp256k1.so /usr/local/lib
+RUN ln -snf /usr/local/lib/libsecp256k1.so /usr/local/lib/libsecp256k1.so.1 && \
+    ln -snf /usr/local/lib/libsecp256k1.so /usr/local/lib/libsecp256k1.so.1.0.1
+RUN ldconfig
+
+# Copy binaries from build stage
+COPY --from=build --chown=root:root /usr/local/bin/cardano-cli /usr/local/bin/cardano-cli
+COPY --from=build --chown=root:root /usr/local/bin/cardano-node /usr/local/bin/cardano-node
+
+# Create cardano group and user
+RUN groupadd --gid 10000 cardano && \
+    useradd --comment 'cardano' --create-home --gid 10000 --password '!' --shell '/bin/bash' --uid 10000 cardano
+
+# Create cardano-node directories
+RUN install --directory --owner=root --group=root --mode=0755 /opt/cardano-node && \
+    install --directory --owner=cardano --group=cardano --mode=0750 /opt/cardano-node/config && \
+    install --directory --owner=cardano --group=cardano --mode=0750 /opt/cardano-node/data && \
+    install --directory --owner=cardano --group=cardano --mode=0750 /opt/cardano-node/data/db && \
+    install --directory --owner=cardano --group=cardano --mode=0750 /opt/cardano-node/log && \
+    install --directory --owner=cardano --group=cardano --mode=0750 /opt/cardano-node/pools
+
+# Copy pool config and key material
+COPY --from=build --chown=cardano:cardano /tmp/testnet/pools /opt/cardano-node/pools
+COPY --from=build --chown=cardano:cardano /tmp/testnet/utxos /opt/cardano-node/utxos
+
+# Copy canary_tx.sh
+COPY --chown=root:root canary_tx.sh /
+RUN chmod 0755 /canary_tx.sh
+
+# Copy cmd.sh
+COPY --chown=root:root cmd.sh /
+RUN chmod 0755 /cmd.sh
+
+USER cardano
+#STOPSIGNAL SIGINT
+
+CMD ["/bin/bash", "/cmd.sh"]
```

Given the script is invoked via `/usr/bin/env` and explicitly requests `bash`, one could simply write 

```suggestion
CMD ["/cmd.sh"]
```

I think?